### ISSUE TITLE
XLA tests have high time-to-signal

### ISSUE THREAD

**Issue by suo** [pytorch/pytorch#78182]
According to https://hud.pytorch.org/metrics, XLA is our "heaviest" job in terms of time-to-signal:
<img width="1247" alt="image" src="https://user-images.githubusercontent.com/1617424/170086449-95184404-7744-4dd5-ba15-20fdb0710776.png">

Our current SLA is 3h, so XLA is not quite violating it at the moment. But it would be good to start optimizing TTS now. Ideally this can be done by optimizing the actual tests, but we can also "shard" the test jobs by running them in parallel on multiple hosts.

cc @seemethere @malfet @pytorch/pytorch-dev-infra @bdhirsh

**Comment by suo** [comment]
On the metrics page you can also see individual job duration, here is build vs. test. Test is a lot longer atm
<img width="824" alt="image" src="https://user-images.githubusercontent.com/1617424/170097173-dd6b99ae-b596-44c3-a0f5-d0a43ace9c83.png">

**Comment by suo** [comment]
sounds good thanks!

**Comment by yeounoh** [comment]
Hey @suo thanks for brining this up, Will take a look. I think we can do something about it [comment](https://github.com/pytorch/pytorch/issues/77286#issuecomment-1125509537). I've been meaning to splitting the current test step into xla build and xla test, which will significantly & directly improve the situation.

**Comment by yeounoh** [comment]
By the way, @suo , `"Our current SLA is 3h, so XLA is not quite violating it at the moment. "` just to confirm, is this per step or per overall workflow?

**Comment by suo** [comment]
Overall workflow. So I'm not really sure how splitting the step into build + test would really change things, as the test job is dependent on the build job.

**Comment by JackCaoG** [comment]
majority of the time is coming from building the tf linked by pytorch/xla, I think we are already use a relatively large vm. I think we currently disabled the build cache. @yeounoh maybe we can have someone in the team to look at how to enable the build cache and see if that would make build any faster?

**Comment by clee2000** [comment]
Has there been any progress on this? we are likely going to move this workflow to trunk if the time doesnt get under 3 hrs

**Comment by JackCaoG** [comment]
Hey @clee2000, what does `move this workflow to trunk` means for XLA test?

**Comment by clee2000** [comment]
It would be moving the `linux-bionic-py3_7-clang8-xla-build` and `linux-bionic-py3_7-clang8-xla-test` jobs from being run on every pull request to being run only on commits that have successfully merged into master.  You would also be able to trigger these jobs to run on a PR by adding the `ciflow/trunk` or the `ciflow/all` labels to the PR

**Comment by yeounoh** [comment]
Hi @clee2000 @JackCaoG we had the cloud cacheing set up before to reduce the time, let me spend some time this week and get back by Thursday.

**Comment by yeounoh** [comment]
> It would be moving the `linux-bionic-py3_7-clang8-xla-build` and `linux-bionic-py3_7-clang8-xla-test` jobs from being run on every pull request to being run only on commits that have successfully merged into master. You would also be able to trigger these jobs to run on a PR by adding the `ciflow/trunk` or the `ciflow/all` labels to the PR

Does that mean we would land a XLA breaking PR to master and revert afterward?

**Comment by clee2000** [comment]
> Does that mean we would land a XLA breaking PR to master and revert afterward? 

I'm not sure what you mean by that, but basically it means that people won't get signal regarding XLA tests on their PRs.  If their PR breaks XLA, it would be found out after they merge into master, at which point we would likely revert it.

**Comment by yeounoh** [comment]
> Hi @clee2000 @JackCaoG we had the cloud cacheing set up before to reduce the time, let me spend some time this week and get back by Thursday.

The build time in the downstream pipeline reduced to 1hr from 2hr. Let me start running it in the upstream cc @clee2000

**Comment by yeounoh** [comment]
Copying the response from https://github.com/pytorch/pytorch/pull/82342#issuecomment-1197760455, 
- the XLA test completed in 1.5 hrs with caching

**Comment by yeounoh** [comment]
The caching should improve TTS, the most recent runs (I tested a few times) finished well under 2hrs. I will close this issue for now, and feel free to reopen another issue if it breaches the SLA again. cc @suo @clee2000 @JackCaoG
